lib/tokenable.rb

```diff
@@ -0,0 +1,8 @@
+module Tokenable
+  extend ActiveSupport::Concern
+
+  def save_token
```

Comment:
I think `generate_token` or something like that would be a better name. `save_token` insinuates that you are providing a token to the method or a token has already been provided.